lib/hanami/router/trie.rb

```diff
@@ -22,32 +22,34 @@ def initialize
       # @api private
       # @since 2.0.0
       def add(route, to, constraints)
-        segments = segments_from(route)
+        # segments = segments_from(route)
+        param_keys = []
         node = @root
 
-        segments.each do |segment|
-          node = node.put(segment)
+        segments_from(route).each do |segment|
+          node = node.put(segment, param_keys)
         end
 
-        node.leaf!(route, to, constraints)
+        node.leaf!(param_keys, to, constraints)
       end
 
       # @api private
       # @since 2.0.0
       def find(path)
-        segments = segments_from(path)
+        # segments = segments_from(path)
```

Comment:
```suggestion
```

Reply:
I'm sure I deleted this line too. Strange.